### ISSUE TITLE
Specify Random Seed and emit seed on failure

### ISSUE THREAD

**Issue by dkowis** [eXparity/exparity-stub#12]
When a test using random stubs fails, it'd be super useful to be able to specify the seed that was used to generate the random stuff. This way the test can be repeated. Especially useful if you've got validation tests on your objects, and only a single run out of many fails. Having the seed would let you recreate the test every time, so that you can fix the bug.

**Comment by stewbis** [comment]
Hi, I'll look at how simple this is to add by moving away from RandomUtils to using the java Random class itself which supports setting the seed and then printing out the random seed at when it's first instantiated

**Comment by dkowis** [comment]
Yeah I was looking at this some. It would require using a no-longer-static implementation, which makes it less convenient and would require a significant amount of refactoring. At least, I think so.

**Comment by stewbis** [comment]
I had a look into it and it's a relatively minor change to use a known seed and log it at the start of a test - a change to pass an instance of Random initialized with a known seed would do the trick. A method can then be exposed to set the seed before the test is re-run or additionally it can consume a property value e.g. org.exparity.stub.random.seed=${the_logged_out_seed_value}

This however won't be sufficient to make it repeatable in all cases - for instance, if a whole suite of tests runs and one fails then a re-run with the same seed may not result in the same value being used in the failing test due to variable order of tests being run, etc.

That being said. adding the ability to use a seed and reporting the seed would open up the possibility of diagnosing the failing random value. I'll look to add the feature to support using and reporting known seeds.

**Comment by dkowis** [comment]
Yeah, also due to threading of the tests. You won't know which test suite has which random. If each test could use its own instance of it, it'd be safe for all the scenarios, but then you'd have to not use static methods. That's why I thought it'd be a significant refactor.
